nndownload/nndownload.py

```diff
@@ -1911,23 +1912,186 @@ def download_thumbnail(session: requests.Session, filename: AnyStr, template_par
 
     output("Finished downloading thumbnail for {0}.\n".format(template_params["id"]), logging.INFO)
 
-
-def download_comments(session: requests.Session, filename: AnyStr, template_params: dict):
+def get_niconico_timestamp(input_time=None):
+    if input_time is None:
+        return int(datetime.now(timezone.utc).timestamp())
+    
+    if isinstance(input_time, (int, float)):
+        return int(input_time)
+    
+    try:
+        # Try ISO format first
+        return int(datetime.strptime(input_time, "%Y-%m-%dT%H:%M:%S%z").timestamp())
+    except ValueError:
+        try:
+            # Try date-only format
+            return int(datetime.strptime(input_time, "%Y-%m-%d").replace(tzinfo=timezone.utc).timestamp())
+        except ValueError:
+            return int(datetime.now(timezone.utc).timestamp())  # Fallback
+
+def download_comments(
+    session: requests.Session,
+    filename: str,
+    template_params: dict,
+):
     """Download the video comments."""
 
+    comment_limit: int = 40
+
     output("Downloading comments for {0}...\n".format(template_params["id"]), logging.INFO)
 
     filename = replace_extension(filename, "comments.json")
 
-    comments_post = COMMENTS_API_POST_DATA.format(template_params["thread_params"], template_params["thread_key"]).replace("\'", "\"").replace(": ", ":").replace(", ", ",")
-    session.options(COMMENTS_API, headers=API_HEADERS) # OPTIONS
-    get_comments_request = session.post(COMMENTS_API, data=comments_post, headers=API_HEADERS)
-    get_comments_request.raise_for_status()
+    comments = {
+        "globalComments": [
+            {
+                "count": 0,
+                "id": ""
+            }
+        ],
+        "threads": [],
+    }
+
+    # Convert date limit to timestamp
+    last_time = get_niconico_timestamp()
+
+    # Process each comment thread
+    for thread in template_params["thread_params"]["targets"]:
+        thread_comments = fetch_comments_modern(
+            session,
+            template_params["id"],
+            template_params["comment_server"],
+            template_params["thread_key"],
+            thread,
+            template_params["thread_params"]["language"],
+            last_time,
+            comment_limit,
+        )
+
+        if thread_comments.get("globalCommentsData"):
+            if thread_comments["globalCommentsData"]["count"] > comments["globalComments"][0]["count"]:
```

Comment:
Not sure what is happening here. It only appends to `comments["threads"]` if the globalComments count goes up? Comments would be helpful